agent/agent_endpoint.go

```diff
@@ -1034,8 +1034,21 @@ func (s *HTTPServer) AgentRegisterService(resp http.ResponseWriter, req *http.Re
 	}
 
 	// Add the service.
-	if err := s.agent.AddService(ns, chkTypes, true, token, ConfigSourceRemote); err != nil {
-		return nil, err
+	removeExistingChecks := false
```

Comment:
"Remove" should be substituted with "replace" in this variable and the query param for consistency with the internal functions.